pytorch docs: add fake_quantize functions documentation

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51748 pytorch docs: add fake_quantize functions documentation**

Adding docs for `fake_quantize_per_tensor_affine` and `fake_quantize_per_channel_affine`
functions.

Note: not documenting `fake_quantize_per_tensor_affine_cachemask` and
`fake_quantize_per_channel_affine_cachemask` since they are implementation details
of `fake_quantize_per_tensor_affine` and `fake_quantize_per_channel_affine`,
and do not need to be exposed to the user at the moment.

Test plan:

Build the docs locally on Mac OS, it looks good

Differential Revision: [D26270514](https://our.internmc.facebook.com/intern/diff/D26270514)

vkuzo committed:
pytorch docs: add fake_quantize functions documentation

Adding docs for `fake_quantize_per_tensor_affine` and `fake_quantize_per_channel_affine`
functions.

Note: not documenting `fake_quantize_per_tensor_affine_cachemask` and
`fake_quantize_per_channel_affine_cachemask` since they are implementation details
of `fake_quantize_per_tensor_affine` and `fake_quantize_per_channel_affine`,
and do not need to be exposed to the user at the moment.

Test plan:

Build the docs locally on Mac OS, it looks good

[ghstack-poisoned]

vkuzo committed:
Update on "pytorch docs: add fake_quantize functions documentation"


Adding docs for `fake_quantize_per_tensor_affine` and `fake_quantize_per_channel_affine`
functions.

Note: not documenting `fake_quantize_per_tensor_affine_cachemask` and
`fake_quantize_per_channel_affine_cachemask` since they are implementation details
of `fake_quantize_per_tensor_affine` and `fake_quantize_per_channel_affine`,
and do not need to be exposed to the user at the moment.

Test plan:

Build the docs locally on Mac OS, it looks good

[ghstack-poisoned]

vkuzo committed:
Update on "pytorch docs: add fake_quantize functions documentation"


Adding docs for `fake_quantize_per_tensor_affine` and `fake_quantize_per_channel_affine`
functions.

Note: not documenting `fake_quantize_per_tensor_affine_cachemask` and
`fake_quantize_per_channel_affine_cachemask` since they are implementation details
of `fake_quantize_per_tensor_affine` and `fake_quantize_per_channel_affine`,
and do not need to be exposed to the user at the moment.

Test plan:

Build the docs locally on Mac OS, it looks good

Differential Revision: [D26270514](https://our.internmc.facebook.com/intern/diff/D26270514)

[ghstack-poisoned]

albanD commented:
Thanks for the doc update!
For the functions that are just c++ helpers, you can add them to https://github.com/pytorch/pytorch/blob/28c5d90b679c6b38bf4183ec99f16d933c2f1bcd/tools/autograd/gen_python_functions.py#L65 to prevent the python binding from being generated for them.

vkuzo committed:
Update on "pytorch docs: add fake_quantize functions documentation"


Adding docs for `fake_quantize_per_tensor_affine` and `fake_quantize_per_channel_affine`
functions.

Note: not documenting `fake_quantize_per_tensor_affine_cachemask` and
`fake_quantize_per_channel_affine_cachemask` since they are implementation details
of `fake_quantize_per_tensor_affine` and `fake_quantize_per_channel_affine`,
and do not need to be exposed to the user at the moment.

Test plan:

Build the docs locally on Mac OS, it looks good

Differential Revision: [D26270514](https://our.internmc.facebook.com/intern/diff/D26270514)

[ghstack-poisoned]

vkuzo commented:
> Thanks for the doc update!
> For the functions that are just c++ helpers, you can add them to
> 
> https://github.com/pytorch/pytorch/blob/28c5d90b679c6b38bf4183ec99f16d933c2f1bcd/tools/autograd/gen_python_functions.py#L65
> 
> to prevent the python binding from being generated for them.

makes sense, thanks!  Will fix that in a follow up PR.
